server/routes/ad.ts

```diff
@@ -333,7 +335,7 @@ const getDetectors = async (
         query_string: {
           fields: ['name', 'description'],
           default_operator: 'AND',
-          query: `*${search.trim().split(' ').join('* *')}*`,
+          query: `*${search.trim().split('-').join('* *')}*`,
```

Comment:
Had to change this line to delimit on `-` rather than ` ` to return results for detectors with `-`s in them. This is because the default tokenizer for ES splits up tokens on `-`.

Example: have some detector called `opendistro-sample-ecommerce-detector`, send an ES search request like `opendistro-sample*` will return empty, since it will be split into searching for `opendistro` AND `sample*`. In this case, `sample*` regex returns empty, so total result returns empty.

Since only other calls to this function send search params of `""`, changing this here doesn't break any existing functionality.

Reply:
I am a little confused: given delimit on `-` instead of ` ` in this line, if input search is `opendistro-sample*`, then the query becomes `*opendistro* *sample**`, will such query returns empty?

From the explanation:
```
Example: have some detector called opendistro-sample-ecommerce-detector, send an ES search request like opendistro-sample* will return empty, since it will be split into searching for opendistro AND sample*. In this case, sample* regex returns empty, so total result returns empty.
```

It looks like changing to `-` as delimiter still returns empty result. 

Btw, can you share any doc regarding `the default tokenizer for ES splits up tokens on  -`?

Reply:
> if input search is `opendistro-sample*`, then the query becomes `*opendistro* *sample**`, will such query returns empty?

This will return results, and is the fixed behavior. Given some detector named `opendistro-sample-ecommerce-detector`, this will match the regex of both `*opendistro*` and `*sample**`.

Sorry for the confusion, but the example I wrote was in the case of splitting by ` ` rather than `-`, and was the problem I was having with returning results originally (I edited it to clarify).

As of 5.1, the default tokenizer was the standard tokenizer, which split based on common punctuation (including `-`). See [here](https://www.elastic.co/guide/en/elasticsearch/reference/5.1/analysis-standard-analyzer.html). There are several threads online of people struggling with searching with terms that include `-`, like [this one](https://discuss.elastic.co/t/how-to-search-for-terms-containing-hyphen-on--all-field/81335).

Reply:
I agree, this could cause some confusion here. Will confirm with UX